### PR TITLE
DVSP-19144: st local client and st remote client can synchronize files

### PR DESCRIPTION
**Pull request by jhalehol**
Allow connections only between two devices, one local CLI and the remove syncthing server (DevSpace)

### CODE REVIEW

**Comment by maruthijr** [comment]
This is a forked open source repo and does not have a CI build in our product. Hence will not have a "green build". Please skip this validation in QE Review

File: cmd/strelaysrv/session.go [hunk 1]
```diff
@@ -128,6 +134,61 @@ func (s *session) AddConnection(conn net.Conn) bool {
...
+	if cidrIpWhiteList == "" {
+		if debug {
+			log.Println("Session doesn't require validate new connection, IP whitelist is empty")
+		}
+		return true
+	}
+
+	switch len(s.conns) {
+	case 2:
+		log.Println("Session", s, "doesn't allow more than two connections peer session")
```
Comment (maruthijr) [inline cmd/strelaysrv/session.go]: Spaces after `Session` and before `doesn't`?

File: cmd/strelaysrv/session.go [hunk 2]
```diff
@@ -128,6 +134,61 @@ func (s *session) AddConnection(conn net.Conn) bool {
 	return false
 }
 
+func (s *session) allowNewConnection(conn net.Conn) bool {
```
Comment (maruthijr) [inline cmd/strelaysrv/session.go]: I am not a `go` expert but it does not appear threadsafe to me. Can you re-check?

File: cmd/strelaysrv/session.go (hunk 1 above)
Comment (jhalehol) [reply cmd/strelaysrv/session.go]: They are not required, it's adding automatically

Comment (jhalehol) [reply cmd/strelaysrv/session.go]: it's not required

File: cmd/strelaysrv/session.go (hunk 2 above)
Comment (jhalehol) [reply cmd/strelaysrv/session.go]: The idea is filtering the session only between two devices as is expected from the specs, that's the point that I saw that is possible to filter it validating the available IPs CIDRs, I am not an expert in go as well, but we tested this and is working as expected, so for this reason, I don't understand your request, to be honest, this code is new for me as well, but we used much time trying to test where I can filter the connections and it's the point to do this

**Pull request merged**